app/models/session.rb

```diff
@@ -0,0 +1,56 @@
+class Session < ApplicationRecord
+  enum status: { 'in-progress': 0, done: 1 }
+  enum session_type: { home: 0, away: 1 }
+
+  belongs_to :user
+
+  before_update :reward
+  after_update_commit :create_wallet_transaction
+
+  def end_session
+    self.end_time = Time.zone.now
+    update!(status: 1)
+  end
+
+  private
+
+  def reward
+    if session_type == 'home'
+      self.rewards = rewards_for_isolation
+    elsif session_type == 'away'
+      self.rewards = rewards_for_being_away
+    end
+  end
+
+  def session_minutes
+    ((end_time - start_time) / 1.minutes).round
+  end
+
+  def rewards_for_isolation
+    (session_minutes / 10).round
+  end
+
+  def rewards_for_being_away
+    -(session_minutes / 10).round * 10
+  end
+
+  def create_wallet_transaction
```

Comment:
Can we move this whole logic into a PORO in lib? session_ender.rb or something like that ? 
This login doesn't really belong here apart from the "helper session_minutes"

Also callbacks can be a pain as the table grows and more logic is added

Reply:
Keeping the persistence and domain layers helps in keeping things clean and adhere to SOLID principles.